ledger-core/virtual/integration/constructor_test.go

```diff
@@ -292,3 +294,128 @@ func TestVirtual_Constructor_NoVFindCallRequestWhenMissing(t *testing.T) {
 
 	mc.Finish()
 }
+
+// A.New calls B.New
+func TestVirtual_CallConstructorFromConstructor(t *testing.T) {
+	t.Log("C5090")
+
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
+	defer server.Stop()
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+		return execution.Request.Callee.String()
+	})
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+
+	var (
+		isolation = contract.ConstructorIsolation()
+
+		classA        = gen.UniqueReference()
+		outgoingA     = server.RandomLocalWithPulse()
+		objectAGlobal = reference.NewSelf(outgoingA)
+
+		classB        = gen.UniqueReference()
+		objectBGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+	)
+
+	// add ExecutionMocks to runnerMock
+	{
+		outgoingCall := executionevent.NewRPCBuilder(gen.UniqueReference(), classA).CallConstructor(classB, "New", []byte("123"))
+		objectAResult := requestresult.New([]byte("finish A.New"), objectAGlobal)
+		objectAResult.SetActivate(reference.Global{}, classA, []byte("state A"))
+		objectAExecutionMock := runnerMock.AddExecutionMock(classA.String())
+		objectAExecutionMock.AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [A.New]")
+				require.Equal(t, classA, ctx.Request.Callee)
+				require.Equal(t, outgoingA, ctx.Request.CallOutgoing)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:     executionupdate.OutgoingCall,
+				Error:    nil,
+				Outgoing: outgoingCall,
+			},
+		)
+		objectAExecutionMock.AddContinue(
+			func(result []byte) {
+				t.Log("ExecutionContinue [A.New]")
+				require.Equal(t, []byte("finish B.New"), result)
+			},
+			&executionupdate.ContractExecutionStateUpdate{
+				Type:   executionupdate.Done,
+				Result: objectAResult,
+			},
+		)
+
+		objectBResult := requestresult.New([]byte("finish B.New"), objectBGlobal)
+		objectBResult.SetActivate(reference.Global{}, classB, []byte("state B"))
+		runnerMock.AddExecutionMock(classB.String()).AddStart(
+			func(ctx execution.Context) {
+				t.Log("ExecutionStart [B.New]")
+				require.Equal(t, classB, ctx.Request.Callee)
+				require.Equal(t, classA, ctx.Request.Caller)
```

Comment:
this is wrong check... ctx.Request.Caller here should be Reference that would get object A after finishing constructor call.

Reply:
execution ctx here from outgoingCall   
```
outgoingCall := executionevent.NewRPCBuilder(gen.UniqueReference(), classA).CallConstructor(classB, "New", []byte("123"))
```
RPCBuilder uses classA, because objectA doesn't exist when we do the creation of outgoingCall
VCallRequest from outgoingCall uses RPCbuilder object as Caller

Reply:
Oh, I get it.  
```
		s.execution.Object = reference.NewSelf(s.Payload.CallOutgoing)
```
in prepareExectuion function before ExecutionStart

Reply:
done